### ISSUE TITLE
Add 'Lines of Code' and 'Fun Facts'

### ISSUE THREAD

**Issue by adiralashiva8** [anuraghazra/github-readme-stats#168]
As a user I want to highlight number of lines I code, and also card for fun facts about my profile

Reference - https://sourcerer.io/adiralashiva8

**Comment by FabioRosado** [comment]
This seems like a pretty fun thing to add. I did a quick look on the Github API docs and it seems there is no way to count lines of commits. After a quick search it seems you can do it with a script that can check the lines of code committed from you, so that might be the only way to implement it 🤔

**Comment by adiralashiva8** [comment]
If not possible we can ignore this request.

**Comment by anuraghazra** [comment]
@adiralashiva8 Thanks for requesting the feature, This seems interesting to implement.

We have to do some research with the  https://sourcerer.io's source code to see if we can salvage some algorithms to calculate the Fun Facts Card.

And @FabioRosado points are totally valid too.

**Comment by MirayXS** [comment]
really like this idea. Interesting!

**Comment by FabioRosado** [comment]
I've tried sourcerer and  you need to download the sourcerer app and choose which repos to add to your profile then it will do some "magic" and will show you, your stats. To add this feature we might need to work on some piece of code that does the same thing 🤔

**Comment by anuraghazra** [comment]
> I've tried sourcerer and you need to download the sourcerer app and choose which repos to add to your profile then it will do some "magic" and will show you, your stats. To add this feature we might need to work on some piece of code that does the same thing

Yes i've also used so https://sourcerer.io/ i think they also use web scraping to collect some extra info which are not in the GraphQL API.

**Comment by xiaoyu2006** [comment]
> > I've tried sourcerer and you need to download the sourcerer app and choose which repos to add to your profile then it will do some "magic" and will show you, your stats. To add this feature we might need to work on some piece of code that does the same thing
> 
> Yes i've also used so https://sourcerer.io/ i think they also use web scraping to collect some extra info which are not in the GraphQL API.

<img width="1429" alt="截屏2020-07-25 19 41 30" src="https://user-images.githubusercontent.com/32544798/88456260-d9af5800-ceae-11ea-96ae-69cd71105576.png">

It's definitely magic. Some simple API can't handle that.

**Comment by anuraghazra** [comment]
> > > I've tried sourcerer and you need to download the sourcerer app and choose which repos to add to your profile then it will do some "magic" and will show you, your stats. To add this feature we might need to work on some piece of code that does the same thing
> > 
> > 
> > Yes i've also used so https://sourcerer.io/ i think they also use web scraping to collect some extra info which are not in the GraphQL API.
> 
> <img alt="截屏2020-07-25 19 41 30" width="1429" src="https://user-images.githubusercontent.com/32544798/88456260-d9af5800-ceae-11ea-96ae-69cd71105576.png">
> 
> It's definitely _magic_. Some simple API can't handle that.

"magic" here is referred to algorithm to calculate those statistics.  https://github.com/sourcerer-io/sourcerer-app/blob/develop/src/main/kotlin/app/FactCodes.kt

*if apps runs on magic i would have started graduating from hogwarts :p*

**Comment by xiaoyu2006** [comment]
> "magic" here is referred to algorithm to calculate those statistics. https://github.com/sourcerer-io/sourcerer-app/blob/develop/src/main/kotlin/app/FactCodes.kt

I know that.

**Comment by xiaoyu2006** [comment]
What about https://stackoverflow.com/a/32154303/10811334.
TL;DR

**Comment by anmol098** [comment]
@adiralashiva8 I think this is what you want [](https://github.com/anmol098/waka-readme-stats/tree/dev#extras )
please have a look at this

**Comment by adiralashiva8** [comment]
@anmol098 Thanks for sharing, It meets my requirement.  If samething implemented here with nice UI, that could be awesome.

**Comment by anmol098** [comment]
@adiralashiva8 We are directly writing on the markdown file...UI  Customization is limited on markdown file.

**Comment by anuraghazra** [comment]
> @adiralashiva8 I think this is what you want https://github.com/anmol098/waka-readme-stats/tree/dev#extras
> please have a look at this

How does it calculates the LOC? it uses WakaTime API (never tried it)?

**Comment by adiralashiva8** [comment]
Thanks for clarification @anmol098 

Seems author is working on feature based on reference i shared, may be this helps them in implementation of feature

Let us wait for @anuraghazra and @FabioRosado view on this (im bit curious about UI part, which makes my profile bit awesome)

**Comment by anuraghazra** [comment]
Actually there are lot of options/ways to calculate total LOC of a github user, but github-readme-stats needs to be realtime & have to calculate them fast (under 10sec limit of vercel) which is not an issue with services which uses Github Action.  

https://sourcerer.io is calculating LOC on demand but as you can see they show a nice little loader & it takes lot of time (they even email you when the calculations are done)

**Comment by xiaoyu2006** [comment]
The problem is not everyone tracks their coding time with WakaTime.
